internal/services/search/search_service_resource.go

```diff
@@ -233,6 +245,11 @@ func resourceSearchServiceCreate(d *pluginsdk.ResourceData, meta interface{}) er
 		return fmt.Errorf("%q SKUs in %q mode can have a maximum of 3 partitions, got %d", string(services.SkuNameStandardThree), string(services.HostingModeHighDensity), partitionCount)
 	}
 
+	// NOTE: Semantic Search SKU cannot be set if the SKU is 'free'
+	if skuName == services.SkuNameFree && semanticSearchSku != string(services.SearchSemanticSearchDisabled) {
+		return fmt.Errorf("'semantic_search_sku' can only be set to %q if the 'sku' field is set to %q, got %q", string(services.SearchSemanticSearchDisabled), string(services.SkuNameFree), semanticSearchSku)
```

Comment:
which allows this to become:

```suggestion
		return fmt.Errorf("`semantic_search_sku` can only be specified when `sku` is not set to %q string(services.SkuNameFree)
```